clarifai/datasets/upload/base.py

```diff
@@ -53,9 +52,15 @@ def get_protos(self, input_ids: List[int]
 class ClarifaiDataLoader:
   """Clarifai data loader base class."""
 
-  def __init__(self, split: str) -> None:
+  def __init__(self) -> None:
     pass
 
+  @property
+  def task(self):
+    raise NotImplementedError("Task should be one of \
+                              'text_classification', 'visual-classification', \
+                              'visual_detection', 'visual_segmentation', 'visual-captioning'")
+
   def load_data(self) -> None:
```

Comment:
Use underscore in all?
Instead of hardcoding, could reference `DATASET_UPLOAD_TASKS`

Reply:
Done